src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,40 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+        int[] salaries = new int[names.length];
+        for (String record : data) {
+            String[] parts = record.split(" ");
+            LocalDate workDate = LocalDate.parse(parts[0], formatter);
+            String employeeName = parts[1];
+            int hoursWorked = Integer.parseInt(parts[2]);
+            int hourlyRate = Integer.parseInt(parts[3]);
+            if (workDate.isBefore(from) || workDate.isAfter(to)) {
+                continue;
+            }
+            for (int i = 0; i < names.length; i++) {
+                if (names[i].equals(employeeName)) {
+                    salaries[i] += hoursWorked * hourlyRate;
+                }
+            }
+        }
+        StringBuilder report = new StringBuilder();
+        report.append("Report for period ")
```

Comment:
```suggestion
        StringBuilder report = new StringBuilder()
                    .append("Report for period ")
```
you can use object right after it's creation too